Samples/Build.Windows.PlatformToolset.props

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8"?>
```

Comment:
I think the name of this file can be improved:
- We *may* want to make this file general-purpose. That way, we can add more helper MSBuild "code" like this in the future if needed.
- It's C++ specific, it'd help clarity-wise to include that in the name. Not everyone looking at the file will know it's C++ specific based on `PlatformToolset`.
- I don't know that it makes sense to say Windows in the name. All our samples are for Windows. Even if this file was `PlatformToolset` specific: `PlatformToolset` doesn't apply to Windows in general, only to UWPs.

Reply:
+1 good suggestions.
I think the name was just copied from my other implementation, so probably that where it originally came from.

Reply:
> doesn't apply to Windows in general, only to UWPs.

Wouldn't we want to set the `PlatformToolset` for all application samples, regardless of UWP or not?

Reply:
I don't know much about this toolset. It's based on what I see in the VS installer that I said that it's UWP specific:
![image](https://user-images.githubusercontent.com/35987549/135692664-84b46e4a-f48c-46c2-a694-246a24202e30.png)
That does then beg the question of whether it's needed in these non-UWPs... @erik0686, can I leave it to you to learn more about this toolset, whether we're needlessly including it in these apps, etc.?

Reply:
I think the "v142" or "v143" for the `PlatformToolset` just refers to the version of the [MSVC compiler/linker/etc.](https://docs.microsoft.com/en-us/cpp/build/how-to-modify-the-target-framework-and-platform-toolset?view=msvc-160#:~:text=The%20platform%20toolset%20consists%20of%20the%20C%2B%2B%20compiler,linker%20%28link.exe%29%2C%20along%20with%20the%20C%2FC%2B%2B%20standard%20libraries.) that get used to build... So it could be used for either UWP or not. I'm not sure why the VS Installer makes it look like they are UWP-only...  🤔

FWIW, I've previously used the same auto-detection logic for "Classic Desktop C++" projects and it worked fine. :)

Reply:
What would you think of a generic name such as Build.Helpers.PlatformToolset.props ??? would that sound as a better name?

Reply:
I looked in "Desktop Development with C++" and I see this:
![image](https://user-images.githubusercontent.com/35987549/135693443-7c988a62-1e26-471b-86d8-0de6531f4535.png)
I don't know if "MSVC v142" is a separate toolset or not...
Assuming they're not, I also don't know which toolset is referenced when you include `<PlatformToolset>v142</PlatformToolset>` in your project file.

Reply:
> Assuming they're not, I also don't know which toolset is referenced when you include <PlatformToolset>v142</PlatformToolset> in your project file.

Correcting something I said. If there are two different toolsets, presumably the right one is selected based on whether the app is a UWP or not.

I think PlatformToolset is automatically added to the project file. So it must be needed.

So I think we're all good here. Regarding the name, Erik, about: Build.Common.Cpp.props?

Reply:
@erik0686 Given all this, it seems to me that the proper way to verify the change would be to: build in VS cmd and look in the binlog to see which compiler was used!

Reply:
FWIW, I like Rohan's suggestion of `Build.Common.Cpp.props` for the file name.

Reply:
Thanks for the suggestions, @rohanp-msft,
For the PlatformToolset, I can see the version is correctly being selected, in the binlog file, I can see this:
![image](https://user-images.githubusercontent.com/26077674/135898674-8fbe27e1-079d-42cd-be59-5cf717d74d03.png)

As for the MSVC Version being used, I'm not exactly sure if this is what you were asking for, but I see this:
![image](https://user-images.githubusercontent.com/26077674/135898773-6abebb66-061a-4f34-aa41-56a45772c064.png)
Does this correctly answer your question? I can recheck for the correct things if they were not.

Reply:
That's it, @erik0686. Thanks. Here's why I think that's right:

I see "The toolset version is v140 for Visual Studio 2015, v141 for 2017, and v142 for 2019." in https://docs.microsoft.com/en-us/cpp/porting/binary-compat-2015-2017?view=msvc-160.

On my machine, I see:
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30423

So I think v\d\d\d is based on the first three digits of the full version number.

So I think 14.29.30133 being used for your selection v142  is right.

Relevant fact that I learned from the link above: I learned that the major number (14) being the same after VS 2013 is to indicate binary compatibility of binaries built using different version 14.* toolsets.

Reply:
Ohhhh that is really cool info, Rohan! thanks for sharing :)